### PR TITLE
Add UI tests for optional app lock setup.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR runs through the create and unlock flows for the app lock feature when presented from settings.

Will follow this PR up with another one for the mandatory flow shown during auth.

### CODE REVIEW

File: ElementX/Sources/Other/AccessibilityIdentifiers.swift [hunk 1]
```diff
@@ -41,6 +45,25 @@ struct A11yIdentifiers {
     static let notificationSettingsScreen = NotificationSettingsScreen()
     static let notificationSettingsEditScreen = NotificationSettingsEditScreen()
     static let createPollScreen = CreatePollScreen()
+    
+    struct AlertInfo {
+        let primaryButton = "alert_info-primary_button"
+        let secondaryButton = "alert_info-secondary_button"
+    }
```
Comment (pixlwave) [inline ElementX/Sources/Other/AccessibilityIdentifiers.swift:52]: I kind of hoped there was a way to detect this from `app.alerts[0]` but I just got a generic array of buttons, so hard-coded these identifiers to fix it.

File: ElementX/Sources/UITests/BlanckFormCoordinator.swift
Comment (pixlwave) [inline ElementX/Sources/UITests/BlanckFormCoordinator.swift:1]: Weird diff, but ok 🙈

File: ElementX/Sources/UITests/UITestsAppCoordinator.swift [hunk 2]
```diff
@@ -42,12 +42,12 @@ class UITestsAppCoordinator: AppCoordinatorProtocol {
     }
     
     func start() {
-        // disabling CA animations
+        // Fix the app tint colour.
         UIApplication.shared.connectedScenes.forEach { scene in
             guard let delegate = scene.delegate as? UIWindowSceneDelegate else {
                 return
             }
-            delegate.window??.layer.speed = 0
```
Comment (pixlwave) [inline ElementX/Sources/UITests/UITestsAppCoordinator.swift:50]: This caused a very specific problem: Presenting a sheet, and then focussing a text field on that sheet, would completely disable all interaction on that sheet. I have no idea why, but running the UI tests on the PR to check if it is needed now that we use `Transaction`.

Comment (pixlwave) [reply ElementX/Sources/UITests/UITestsAppCoordinator.swift:50]: The run is here: https://github.com/vector-im/element-x-ios/actions/runs/6798348298

File: UITests/Sources/AppLockSetupUITests.swift [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+import XCTest
+
+@testable import ElementX
+
+@MainActor
+class AppLockSetupUITests: XCTestCase {
+    var app: XCUIApplication!
+    
+    @MainActor enum Step {
```
Comment (alfogrillo) [inline UITests/Sources/AppLockSetupUITests.swift:25]: Is `@MainActor` really needed here for the `Step`?

**Review by alfogrillo — APPROVED** [review]
LGTM 👏🏻

File: UITests/Sources/AppLockSetupUITests.swift (hunk 3 above)
Comment (pixlwave) [reply UITests/Sources/AppLockSetupUITests.swift:25]: Yeah its stupid: [`UIDevice`](https://developer.apple.com/documentation/uikit/uidevice) is MainActor so to keep the `isPhone` property synchronous (and therefore the 2 change pin properties), this was the simplest way to do it.

Comment (alfogrillo) [reply UITests/Sources/AppLockSetupUITests.swift:25]: Maybe you could have moved the annotation just to that property (`isPhone`) to make this clear, but it doesn't really make any difference I guess.

File: ElementX/Sources/UITests/UITestsAppCoordinator.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/UITests/UITestsAppCoordinator.swift:50]: Seems to have worked.

**Pull request merged**